ui/src/selection-panel/track-number-panel/color-selector/color-selector-utils.ts

```diff
@@ -26,5 +26,5 @@ export const getColors = () => {
 
 export const getDefaultColorKey = (id: LayoutTrackNumberId): TrackNumberColorKey => {
     const colors = getColors();
-    return getUnsafe(colors[parseInt(id.replace(/^\D+/g, '')) % colors.length])[0];
+    return expectDefined(colors[parseInt(id.replace(/^\D+/g, '')) % colors.length])[0];
```

Comment:
Aika kompleksi rivi joka ehkä pitäisi hajotella, tässäkin voisi toki käyttää vielä sitä first()-funkkaria.

Reply:
Refakkasin tämän koko jutun vähän jakamalla sitä osiin. Tässä tosin näyttää siltä että `first()` olisi oikea työkalu, mutta oikeasti tuo indeksoitava juttu on `[A, B]`-tyyppinen tuple. `first()` on tarkoitettu taulukoille eikä tupleja ymmärrä, ja siten se palauttaisi alkion tyyppiä `A | B | undefined`. En viitsi tosin lähteä muuttamaan `first()`:in toimintaa tämän osalta, sillä mielestäni se ei ole näissä tapauksissa muutenkaan oikea työkalu.